tests/test_filter_count.py

```diff
@@ -181,43 +270,99 @@ def test_prune_on_small_data(filter_by):
     assert all(calibrated_cj.sum(axis=1).round().astype(int) == label_counts)
-    assert len(data["labels"]) == int(round(np.sum(calibrated_cj)))
+    assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj)))
 
     calibrated_cj2 = count.compute_confident_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
         calibrate=True,
     )
 
     # Check equivalency
     assert np.all(calibrated_cj == calibrated_cj2)
 
 
+def test_calibrate_joint_multilabel():
+    dataset = multilabel_data
+    cj = count.compute_confident_joint(
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        multi_label=True,
+        calibrate=False,
+    )
+    calibrated_cj = count.calibrate_confident_joint(
+        confident_joint=cj,
+        labels=dataset["labels"],
+        multi_label=True,
+    )
+    y_one = int2onehot(dataset["labels"])
+    # Check calibration
+    for class_num in range(0, len(calibrated_cj)):
+        label_counts = np.bincount(y_one[:, class_num])
+        assert all(calibrated_cj[class_num].sum(axis=1).round().astype(int) == label_counts)
+        assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj[class_num])))
+    calibrated_cj2 = count.compute_confident_joint(
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        calibrate=True,
+        multi_label=True,
+    )
+    # Check equivalency
+    assert np.all(calibrated_cj == calibrated_cj2)
+
+
 @pytest.mark.parametrize("use_confident_joint", [True, False])
 def test_estimate_joint(use_confident_joint):
+    dataset = data
     joint = count.estimate_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
-        confident_joint=data["cj"] if use_confident_joint else None,
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        confident_joint=dataset["cj"] if use_confident_joint else None,
     )
 
     # Check that joint sums to 1.
     assert abs(np.sum(joint) - 1.0) < 1e-6
 
 
+@pytest.mark.parametrize("use_confident_joint", [True, False])
+def test_estimate_joint_multilabel(use_confident_joint):
+    dataset = multilabel_data
+    if use_confident_joint:
+        cj = count.compute_confident_joint(
+            labels=dataset["labels"], pred_probs=dataset["pred_probs"], multi_label=True
+        )
+        joint = count.estimate_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            confident_joint=cj,
+            multi_label=True,
+        )
+    else:
+
+        joint = count.estimate_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            confident_joint=dataset["cj"] if use_confident_joint else None,
+            multi_label=True,
+        )
+    # Check that joint sums to 1.
+    assert abs(np.sum(joint) - 1.0) < 1e-6
```

Comment:
Should't the joint now be K x 2 x 2 in multilabel settings?  I think you forgot to update the `estimate_joint` function?  Please make sure every multilabel occurrence of `confident_joint` in count.py, filter.py is always K x 2 x 2 now.

Reply:
in fact your test should explicitly check that the shape returned by all new multilabel confident joint functions is always: K x 2 x 2

Reply:
added all tests.